linux_os/guide/system/accounts/accounts-session/accounts_users_home_files_groupownership/bash/shared.sh

```diff
@@ -0,0 +1,7 @@
+# platform = multi_platform_all
+# reboot = false
+# strategy = restrict
+# complexity = low
+# disruption = low
+
+awk -F':' '{ if ($4 >= {{{ gid_min }}} && $4 != 65534) system("chgrp -Rf " $4" "$6) }' /etc/passwd
```

Comment:
So if I understand correctly this is changing the group owner of all files in the user's home directory to his own user group.
But the rule is not about all files in a users home being owned by his own group, it is about all files being group owned by a group which the user is a member of, it doesn't need to be his own group.

Ref: https://vaulted.io/library/disa-stigs-srgs/red_hat_enterprise_linux_7_security_technical_implementation_guide/V-204472?version=V3R4

Reply:
Hi @yuumasato , looks the rule you referred is written this way because there is an exception for primary groups in some distros. For example, since RHEL and Ubuntu, by default, automatically create a new group with the same name of the respective new user and set it as primary group for that user, the SLES default approach is to include new users in a shared group called `users`. This rule covers these two approaches since it relies on the primary group informed on `/etc/passwd`.

This also makes sense in a context of other rules from the same benchmark, like:
https://vaulted.io/library/disa-stigs-srgs/red_hat_enterprise_linux_7_security_technical_implementation_guide/V-204470?version=V3R4
https://vaulted.io/library/disa-stigs-srgs/red_hat_enterprise_linux_7_security_technical_implementation_guide/V-204473?version=V3R4

So, in an environment in compliance with these three rules (204470, 204472 and 204473), for example, from the same benchmark means that only the owner can `write` on files under its home directory (makes sense). In this environment it is also possible that other users may `read` files under its home directory (despite the impact on confidentiality, makes sense in some environments, as mentioned above). In this case, even if the group-owner of files under a home directory are different from the primary group, they wouldn't be accessible due the rules `204470` and `204473`. For that make sense the permissions and/or ownership of the home directory itself should also be changed, conflicting with other rules from the same benchmark.

In this context I defend the interpretation that the files should be group-owned by the user primary group and the description is only confused due to the case of distros which share the primary group among regular users. Otherwise, the other two rules (maybe more) should be updated in the benchmark description. But this would clearly impact in less security for the system and doesn't look to be the intention.

Reply:
I agree with you, thank you for the detailed explanation.